addOns/scripts/src/main/java/org/zaproxy/zap/extension/scripts/dialogs/NewScriptDialog.java

```diff
@@ -280,7 +280,11 @@ public String validateFields() {
         }
         if (this.isEmptyField(FIELD_TEMPLATE)) {
             // No template, check the engine supports this
-            if (!getSelectedEngine().isSupportsMissingTemplates()) {
+            ScriptEngineWrapper selectedEngine = getSelectedEngine();
+            if (selectedEngine == null) {
```

Comment:
Is this possible? (I read the commit message that's theoretical but if it's not possible better suppress the FP than change the code.)

Reply:
It "shouldn't" be. However when there are multiple engines then the combo does default to a blank entry. I couldn't find a way to get both the scan engine to be blank and the template, but that may just mean I didn't hit the right combo or that a different platform (Mac) might not behave differently.

Reply:
Run ZAP with Java 15+ and without any script engine add-on, that should leave engine and template empty. Previous validation ensures there's a template or engine selected.

Reply:
Tested with Java 16, will revert and mark as FP in SonarCloud.